### PR TITLE
Add Data.Set to Prolude

### PR DESCRIPTION
**Pull request by saralich**
Story details: https://app.clubhouse.io/itprotv/story/51968

### CODE REVIEW

File: Prolude/Foldable.hs [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
 import Data.Map (Map)
+import Data.Set (Set)
 import GHC.Exts (fromList)
+import qualified Data.Bool as Bool
+import qualified Data.Foldable as Foldable
+import qualified Data.Ord as Ord
+import qualified Data.Set as Set
+
+emptySet :: Set a
+emptySet = Set.empty
```
Comment (gera-cameron) [inline Prolude/Foldable.hs:25]: mempty?

**Comment by tfausak** [comment]
I feel pretty much the same as #5: I like exporting `Set`, but for everything else I prefer the qualified version (`Set.insert`) over the prefixed version (`setInsert`).

File: Prolude/Foldable.hs [hunk 2]
```diff
@@ -1,10 +1,40 @@
...
+setMember = Set.member
+
+setSingleton :: a -> Set a
+setSingleton = Set.singleton
+
+setUnion :: Ord.Ord a => Set a -> Set a -> Set a
+setUnion = Set.union
+
+setUnions :: (Foldable.Foldable f, Ord.Ord a) => f (Set a) -> Set a
+setUnions = Set.unions
```
Comment (gera-cameron) [inline Prolude/Foldable.hs:40]: I'm honestly fine with still prefixing all of these.

**Comment by gera-cameron** [comment]
> I feel pretty much the same as #5: I like exporting `Set`, but for everything else I prefer the qualified version (`Set.insert`) over the prefixed version (`setInsert`).

Ditto

**Review by gera-cameron — APPROVED** [review]
:taco:

**Review by tfausak — APPROVED** [review]

**Pull request merged**